src/main/java/seedu/address/logic/parser/LoginCommandParser.java

```diff
@@ -42,4 +45,28 @@ public LoginCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, LoginCommand.MESSAGE_USAGE));
         }
     }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the LoginCommand
+     * and returns an LoginCommand object for execution.
+     * This method is used to parse arguments for GUI login.
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public LoginCommand parse(String enteredUsername, String enteredPassword) throws ParseException {
+        requireNonNull(enteredUsername);
+        requireNonNull(enteredPassword);
+        if (enteredUsername.isEmpty()) {
+            throw new ParseException(EMPTY_USERNAME_MESSAGE);
+        }
+        if (enteredPassword.isEmpty()) {
+            throw new ParseException(EMPTY_PASSWORD_MESSAGE);
```

Comment:
I went into your branch to test, but I can't seem to get either of these messages to display

Reply:
added tests for it